compiler-core/src/type_/tests/errors.rs

```diff
@@ -1160,6 +1160,33 @@ fn const_annotation_wrong_4() {
     assert_module_error!("pub const pair: #(Int, Float) = #(4.1, 1)");
 }
 
+#[test]
+fn const_annotation_wrong_5() {
+    assert_module_error!(
+        "const mismatched_types: String = 7
+const invalid_annotation: MyInvalidType = \"str\"
+const invalid_value: String = MyInvalidValue
+const invalid_unannotated_value = [1, 2.0]
+const invalid_everything: MyInvalidType = MyInvalidValue"
+    );
+}
+
+#[test]
+fn const_annotation_wrong_6() {
+    assert_module_error!(
+        "const num: String = 7
+const str1: MyInvalidType = \"str\"
+const tpl: String = #(Ok(1), MyInvalidType, 3)
+const lst = [1, 2.0]
+const unbound: MyInvalidType = MyInvalidType
+const assignment1: String = num
+const assignment2: String = str1
+const assignment3: String = tpl
+const assignment4: String = lst
+const assignment5: String = unbound"
+    );
+}
```

Comment:
What are these tests testing? There's a lot happening and I can't tell what the intent is. I don't think I would be able to tell if it is a bug or not if the output of these changes.

Instead we should have many small tests, each testing precisely one edge case, ideally with a comment or descriptive name to state what is being tested.